app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -299,7 +304,7 @@ class BrowserTabViewModel(
 
     fun onViewVisible() {
         command.value = if (!currentBrowserViewState().browserShowing) ShowKeyboard else HideKeyboard
-        ctaViewModel.refreshCta()
+        refreshCta(true)
```

Comment:
`currentBrowserViewState().browserShowing` only tells us if the browser is showing on the fragment which is always true once you navigate anywhere, however there are multiple calls happening even if the tab fragment is not visible triggering multiple dialogs. Ideally we want to know when the user is seeing a new tab or a browser and if that's the current fragment visible. I played around adding a new `isVisible` attribute in the state but we still get multiple calls in view models that are not visible. I definitely want to have another look at this if the experiment is successful because there is a lot of room for improvement. For now, hardcoding the values works well and doesn't involve changing any more logic.

Reply:
@subsymbolic FYI this one of the areas where there is room for improvement

Reply:
Agreed! I've created this task for it https://app.asana.com/0/1125189844152671/1154748750174826